modules/rarity/rarity.gd

```diff
@@ -0,0 +1,53 @@
+extends Node
+
+
+#region Public Variables
+var rarities: Array[StringName] = []
+var rarity_colors: Dictionary = {}
+#endregion
+
+
+#region Internal Functions
+func _ready() -> void:
+	Modules.register_hooks(handler)
+	
+	register_rarity(&"Free", Color.WHITE)
+	register_rarity(&"Common", Color.GRAY)
+	register_rarity(&"Rare", Color.BLUE)
+	register_rarity(&"Epic", Color.PURPLE)
+	register_rarity(&"Legendary", Color.GOLD)
+#endregion
+
+
+#region Public Functions
+func handler(what: StringName, info: Array) -> bool:
+	if what == &"Update Card":
+		return update_card_hook.callv(info)
+	
+	return true
+
+
+func update_card_hook(card: Card) -> bool:
+	# Rarity Color
+	var rarity_node: MeshInstance3D = card.get_node("Mesh/Rarity")
+	
+	if not card.modules.get("rarities"):
+		assert(false, "'%s' (%d) doesn't have a rarity." % [card.name, card.id])
```

Comment:
Is it a good idea to warn about not using a module? I guess using a disabled module isn't really a problem, so it should be fine.